### PR TITLE
fix(core): prevent infinite loops in clobbered elements check

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit updates HTML sanitization logic to avoid infinite loops in case clobbered elements contain fields like `nextSibling` or `parentNode`. Those fields are used for DOM traversal and this update makes sure that those calls return valid results.

Also this commit fixes an issue when clobbering `nodeName` causes JS exceptions.

(more context in the internal ticket: [b/323800512](http://b/323800512))

## PR Type
What kind of change does this PR introduce?

- [x] Fix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/test/sanitization/html_sanitizer_spec.ts [hunk 1]
```diff
@@ -206,17 +206,59 @@ describe('HTML sanitizer', () => {
...
+      // depending on the browser, we might get an exception
+    }
+  });
+
+  it('should properly sanitize the content when `nodeName` is clobbered', () => {
+    const output = sanitizeHtml(defaultDoc, '<form><input name=nodeName></form>text');
+    expect(output).toBe('text');
+  });
+
+  fit('should sanitize the content when `nextSibling` or `firstChild` were clobbered', () => {
```
Comment (alfaproject) [inline packages/core/test/sanitization/html_sanitizer_spec.ts:233]: `fit`

**Comment by AndrewKushnir** [comment]
[Exploratory presubmit](https://test.corp.google.com/OCL:610968919:BASE:610968936:1709096387693:47f24f9c).

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: fw-core 
Reviewed-for: fw-security

**Comment by AndrewKushnir** [comment]
**Caretaker notes**

- [TGP is "green"](https://fusion2.corp.google.com/presubmit/612681455/OCL:612681455:BASE:612868698:1709659674947:5d2af964/targets) for the changes in this PR
- Landing this PR in g3 would require including [this extra change](http://cl/611281955) into the sync
- Please merge and sync this change separately from other changes

**Comment by atscott** [comment]
This PR was merged into the repository by commit eaff724b7746f9df35e69ed8fa1e78c188ebc14c.

**Pull request closed without merging**